src/conlink/core.cljs

```diff
@@ -104,12 +109,14 @@ General Options:
   - Add default values to a link:
     - type: veth
     - dev: eth0
+    - mac: random MAC starting with first octet of 'c2'
     - mtu: --default-mtu (for non *vlan type)
     - base: :conlink for veth type, :host for *vlan types, :local otherwise"
   [{:as link :keys [type base bridge ip route forward]} bridges]
   (let [{:keys [default-mtu docker-eth0? docker-eth0-address]} @ctx
         type (keyword (or type "veth"))
         dev (get link :dev "eth0")
+        mac (get link :mac (random-mac))
```

Comment:
In case you were wondering, given 5 free octets to randomize, there is a 1 in ~ 10^2639 chance of duplicating MACs on the same bridge with 1024 links (less than that, if some of those links are created by scaling containers and in turn scaling the randomly assigned MAC).  I think this means we have ~0 chance of collisions even after a few million runs with fully loaded bridges.

Reply:
Haha, yes, good to know. I decided that hard-coding the first octet to make it more clear that it's conlink assigned was worth the slight increase in risk of collision (veth links already assign random MAC addresses unless you specify them). Some other tools do a similar thing. For example, QEMU and libvirt use '52:54:00', although there might be some mechanism to reduce the likelihood of collisions (some sort of state maintained somewhere).